### PR TITLE
Add codemagic yaml

### PR DESCRIPTION
**Pull request by fran-tirapu**
Workflow to deploy to production and clean cache added.
Production site: https://labs.codemagic.io/

### CODE REVIEW

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]
@maks you can add a command to generate the site before deploying it to S3

**Comment by mikhail-tokarev** [comment]
- [ ] @maks replace plausible code with proper one (see Slack thread)

File: codemagic.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        name: Deploy to production
+        environment:
+            groups:
+                - aws-production
+                - production
+        scripts:
+            - name: Deploy to production
+              script: |
+                  set -e
+                  aws s3 sync ./your-first-flutter-app-to-appstore s3://labs.codemagic.io --delete --cache-control max-age=31536000,public
```
Comment (artemii-yanushevskyi) [inline codemagic.yaml]: I am not sure the intention was to publish this page as a main page
```suggestion
                  aws s3 sync . s3://labs.codemagic.io --delete --cache-control max-age=31536000,public
```

File: codemagic.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+workflows:
+    production:
```
Comment (artemii-yanushevskyi) [inline codemagic.yaml]: It's widely accepted to use 2-space indentation for YAML files

File: codemagic.yaml (hunk 1 above)
Comment (artemii-yanushevskyi) [inline codemagic.yaml]: I believe no need for `--cache-control max-age=31536000,public`

File: codemagic.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        environment:
+            groups:
+                - aws-production
+                - production
+        scripts:
+            - name: Deploy to production
+              script: |
+                  set -e
+                  aws s3 sync ./your-first-flutter-app-to-appstore s3://labs.codemagic.io --delete --cache-control max-age=31536000,public
+                  aws cloudfront create-invalidation --distribution-id $DISTRIBUTION_ID --paths '/*'
```
Comment (artemii-yanushevskyi) [inline codemagic.yaml]: Consider invalidating only the paths that are actually present
```sh
INVALIDATE_PATHS=( $(aws s3 ls --recursive s3://${S3_BUCKET:-$DOMAIN_NAME} | awk '{print "/"$4}' | sed -e 's/index.html
aws cloudfront create-invalidation --distribution-id ${CLOUDFRONT_ID?} --paths ${INVALIDATE_PATHS[@]}
```

**Review by artemii-yanushevskyi — CHANGES_REQUESTED** [review]

File: codemagic.yaml (hunk 2 above)
Comment (mikhail-tokarev) [reply codemagic.yaml]: welcome to 2 or 4-space holy war club :D

File: codemagic.yaml (hunk 1 above)
Comment (mikhail-tokarev) [reply codemagic.yaml]: it's needed

File: codemagic.yaml (hunk 3 above)
Comment (mikhail-tokarev) [reply codemagic.yaml]: in this case better to keep it simple

File: codemagic.yaml (hunk 1 above)
Comment (fran-tirapu) [reply codemagic.yaml]: It is not a page, it is a [folder](https://github.com/codemagic-ci-cd/codelabs) with the index.html file

**Comment by maks** [comment]
> @maks you can add a command to generate the site before deploying it to S3

Thanks @mikhail-tokarev @fran-tirapu and @artemii-yanushevskyi for getting this setup so quickly, much appreciated!

@mikhail-tokarev yes, good point! now done. Following Google's recommended workflow, I first exported the google doc to markdown+assets, committed into this repo and then add script step to build html from that markdown, as going forward we should only edit the markdown if any content updates are needed.
I also needed to switch builder instance to Linux to avoid hassles of codesigning and `claat` tool not having a arm64 macos release build available yet.

**Comment by maks** [comment]
@mikhail-tokarev I moved the url in #[be655b2](https://github.com/codemagic-ci-cd/codelabs/pull/1/commits/be655b23dfdc70b2a9bd1fa14ba5640030551cf2) to allow for future additional codelabs though at the moment I don't have access to the real Codemagic project to test if that deploys correctly.

**Comment by mikhail-tokarev** [comment]
> @mikhail-tokarev I moved the url in #[be655b2](https://github.com/codemagic-ci-cd/codelabs/pull/1/commits/be655b23dfdc70b2a9bd1fa14ba5640030551cf2) to allow for future additional codelabs though at the moment I don't have access to the real Codemagic project to test if that deploys correctly.

I see you have access to `Codemagic Marketing` and I think you deployed blog several times so you do have access

**Comment by maks** [comment]
> > @mikhail-tokarev I moved the url in #[be655b2](https://github.com/codemagic-ci-cd/codelabs/pull/1/commits/be655b23dfdc70b2a9bd1fa14ba5640030551cf2) to allow for future additional codelabs though at the moment I don't have access to the real Codemagic project to test if that deploys correctly.
> 
> I see you have access to `Codemagic Marketing` and I think you deployed blog several times so you do have access

Ah got it, sorry my fault @mikhail-tokarev , I thought it might need to be moved to a different account, but yes all good as yes it has deployed successfully already.

**Comment by maks** [comment]
I think all the feedback has been addressed and publishing workflow seems to work well, so I think this can be merged now.

**Pull request merged**